src/constraints.jl

```diff
@@ -1344,3 +1344,107 @@ function all_constraints(
     append!(ret, all_nonlinear_constraints(model))
     return ret
 end
+
+"""
+    penalty_relaxation(
+        model::Model,
+        [penalties::Dict];
+        [default::Union{Nothing,Real} = nothing,]
+    )
+
+Destructively modify the model in-place to create a penalized relaxation of the
+constraints.
+
+!!! warning
+    This is a destructive routine that modifies the model in-place. If you don't
+    want to modify the original model, use [`copy_model`](@ref) to create a copy
+    before calling [`penalty_relaxation`](@ref).
+
+## Reformulation
+
+See [`MOI.Utilities.ScalarPenaltyRelaxation`](@ref) for details of the
+reformulation.
+
+For each constraint `ci`, the penalty passed to
+[`MOI.Utilities.ScalarPenaltyRelaxation`](@ref) is `get(penalties, ci, default)`.
+If the value is `nothing`, because `ci` does not exist in `penalties` and
```

Comment:
What if the user puts `nothing` as a value in the dictionary ?

Reply:
They can't. I'll update the type signature.